### DISCUSSION TITLE
IAM auth problem

### DISCUSSION THREAD

**Discussion by Shmalikov** [rajanadar/VaultSharp#265] [Q&A]
I'm trying to use IAM auth as written here https://github.com/rajanadar/VaultSharp#aws-auth-method---iam
Also, I'm setting to ignore the validation self-signed certificate:
```
    vaultClientSettings.MyHttpClientProviderFunc = (handler) =>
    {
        var clientHandler = (HttpClientHandler) handler;
        clientHandler.ServerCertificateCustomValidationCallback
            += (message, certificate2, arg3, arg4) => true;
        return new HttpClient(clientHandler);
    };
    var vaultClient = new VaultClient(vaultClientSettings);
```
I'm launching code in AWS Instance, so for credentials I use:
`var awsCredentials = new InstanceProfileAWSCredentials();`
It looks like it should work, awsCredentials is getting all needed stuff **Token, AccessKey, SecretKey.**
But I'm getting the error:
**`VaultSharp.Core.VaultApiException: {"errors":["invalid request header: X-Amz-Content-Sha256"]}`**

What could be the reason for that error?

**Comment by rajanadar** [comment]
hi @Shmalikov 

Do you still have this problem?
